app/tests/spec/models/auth_brokers/first-run.js

```diff
@@ -97,15 +97,27 @@ define(function (require, exports, module) {
     });
 
     describe('beforeSignUpConfirmationPoll', function () {
-      it('notifies the iframe channel, does not halt', function () {
+      var result;
+      beforeEach(function () {
         sinon.spy(iframeChannel, 'send');
 
+        account.set('needsOptedInToMarketingEmail', true);
+
         return broker.beforeSignUpConfirmationPoll(account)
-          .then(function (result) {
-            assert.isTrue(iframeChannel.send.calledWith(broker._iframeCommands.SIGNUP_MUST_VERIFY));
-            assert.isUndefined(result.halt);
+          .then(function (_result) {
+            result = _result;
           });
       });
+
+      it('notifies the iframe channel', function () {
+        assert.isTrue(
+          iframeChannel.send.calledWith(
+            broker._iframeCommands.SIGNUP_MUST_VERIFY, { emailOptIn: true }));
```

Comment:
Does `emailOptIn` default to false or true if not specified or is not boolean? Minor nit would be to add cases to cover these.